homeassistant/components/qnap_qsw/config_flow.py

```diff
@@ -78,6 +78,9 @@ async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowRes
 
         _LOGGER.debug("DHCP discovery detected QSW: %s", self._discovered_mac)
 
+        await self.async_set_unique_id(format_mac(self._discovered_mac))
+        self._abort_if_unique_id_configured()
```

Comment:
Should we update the url if the ip changes? with `updates=`?

Reply:
> Should we update the url if the ip changes? with `updates=`?

I'm not sure that we should do that. What happens if the user has manually configured it with a local hostname such as `http://qnapsw.lan/`?
If that's the case and the IP isn't static we will always be changing the IP without any need to do that since the local hostname is already resolving to the real IP address, right?

Reply:
You could check the entry and only update it if its an ip address and not a domain